packages/legacy-credentials/src/vcInterop.ts

```diff
@@ -0,0 +1,180 @@
+/**
+ * Copyright (c) 2018-2023, BOTLabs GmbH.
+ *
+ * This source code is licensed under the BSD 4-Clause "Original" license
+ * found in the LICENSE file in the root directory of this source tree.
+ */
+
+import {
+  hexToU8a,
+  stringToU8a,
+  u8aCmp,
+  u8aToHex,
+  u8aToString,
+} from '@polkadot/util'
+import { base58Decode, base58Encode, blake2AsU8a } from '@polkadot/util-crypto'
+
+import { CType } from '@kiltprotocol/core'
+import { KiltCredentialV1, Types, constants } from '@kiltprotocol/vc-export'
+
+import type { ICType, IClaim, ICredential } from '@kiltprotocol/types'
+
+import { makeStatementsJsonLD } from './utils.js'
+
+/**
+ * Produces an instance of [[KiltAttestationProofV1]] from an [[ICredential]].
+ *
+ * @param credential Input credential.
+ * @param opts Additional parameters required for creating a proof from an [[ICredential]].
+ * @param opts.blockHash Hash of a block at which the proof must be verifiable.
+ * @returns An embedded proof for a verifiable credential derived from the input.
+ */
+export function proofFromICredential(
```

Comment:
Maybe?
```suggestion
export function vcProofFromICredential(
```

Reply:
maybe? It's not public though, just an internal